service/storage_proxy.cc

```diff
@@ -3797,9 +3797,14 @@ mutation storage_proxy::do_get_batchlog_mutation_for(schema_ptr schema, const st
 template<typename Range>
 bool storage_proxy::cannot_hint(const Range& targets, db::write_type type) const {
     // if hints are disabled we "can always hint" since there's going to be no hint generated in this case
-    return hints_enabled(type) &&
-            _hints_manager.started() &&
-            boost::algorithm::any_of(targets, std::bind(&db::hints::manager::too_many_in_flight_hints_for, &_hints_manager, std::placeholders::_1));
+    return hints_enabled(type) && _hints_manager.started() && std::invoke([&] {
+        for (const auto& target : targets) {
```

Comment:
This is a purely subjective change and to me the `any_of` version is easier and faster to read and understand.

(It could be simplified by replacing `bind` with `bind_front`:
`any_of(targets, std::bind_front(&db::hints::manager::too_many_in_flight_hints_for, &_hints_manager))`
But the `for` + `if` version? It's definitely more clunky than the previous one...
BTW. I recommend this classic talk on the topic https://www.youtube.com/watch?v=W2tWOdzgXHA
section at 2:15 to 33:30 "no raw loops"
)

Anyway such subjective changes should not take place.